api/envoy/api/v2/cds.proto

```diff
@@ -632,6 +667,77 @@ message Cluster {
   // If this flag is not set to true, Envoy will wait until the hosts fail active health
   // checking before removing it from the cluster.
   bool drain_connections_on_host_removal = 32;
+
+  // New mechanism for LB policy configuration. If set and understood
+  // by the client, replaces the
+  // :ref:`lb_policy<envoy_api_field_Cluster.lb_policy>` field, the
+  // :ref:`common_lb_config<envoy_api_field_Cluster.common_lb_config>`
+  // field, and the fields in the
+  // :ref:`lb_config<envoy_api_field_Cluster.lb_config>` oneof. For backward
+  // compatibility, management servers should populate both sets of fields.
+  LoadBalancingPolicy load_balancing_policy = 40;
+}
+
+// Load balancing policy selection.
+//
+// Every LB policy defined via this mechanism will be identified via a
+// unique name using reverse DNS notation. If the policy needs configuration
+// parameters, it must define a message for its own configuration, which will
+// be stored in the config field. The name of the policy will tell clients
+// which type of message they should expect to see in the config field.
+//
+// Envoy supports the following LB policies, each of which has its own
+// configuration message. Refer to
+// :ref:`load balancer type <arch_overview_load_balancing_types>` architecture
+// overview section for information on each type.
+//
+// "io.envoyproxy.weighted_round_robin":
```

Comment:
Nit: these should be `envoy.<foo>`. We cheat for Envoy internals (surprise) and don't do full reverse DNS. The idea is folks should do as we say and not as we do.

Reply:
As we discussed this morning, I've changed this such that instead of deprecating the old fields, we trigger use of the new fields via a new enum value in the `lb_policy` field.  This seems like a smoother plan for adding this into v2, and we can fully transition to the new mechanism in v3.

As a result, this comment has gone away.